src/components/RichSelectField/index.tsx

```diff
@@ -0,0 +1,183 @@
+import { RichSelect } from '@scaleway/ui'
+import { FieldState } from 'final-form'
+import React, {
+  Children,
+  ComponentProps,
+  ReactElement,
+  useCallback,
+  useMemo,
+} from 'react'
+import { useField, useFormState } from 'react-final-form'
+import pickValidators from '../../helpers/pickValidators'
+import useValidation from '../../hooks/useValidation'
+import { useErrors } from '../../providers/ErrorContext'
+import { BaseFieldProps } from '../../types'
+
+type RichSelectProps = ComponentProps<typeof RichSelect>
+type RichSelectOptionProps = ComponentProps<typeof RichSelect.Option>
+type RichSelectOptionElement = ReactElement<RichSelectOptionProps>
+type RichSelectOptions = ComponentProps<typeof RichSelect>['options']
+type RichSelectOptionOrGroup = NonNullable<RichSelectOptions>[number]
+type RichSelectOption = { value: string; label: string }
+
+export type RichSelectFieldProps<
+  T extends RichSelectOptionOrGroup = RichSelectOptionOrGroup,
+> = BaseFieldProps<T> & {
```

Comment:
Can we use the type directly from RichSelect to avoid retype them here ? 

An exemple on CheckboxField

```js
type CheckboxFieldProps<T = CheckboxValue, K = string> = BaseFieldProps<T, K> &
  Partial<
    Pick<
      ComponentProps<typeof Checkbox>,
      | 'checked'
      | 'disabled'
      | 'id'
      | 'onBlur'
      | 'onChange'
      | 'onFocus'
      | 'progress'
      | 'readOnly'
      | 'required'
      | 'size'
      | 'typographyVariant'
      | 'valid'
    >
  > & {
    name: string
    label?: string
    className?: string
    children?: ReactNode
  }
```